dc-cudami-server/dc-cudami-server-business/src/main/java/de/digitalcollections/cudami/server/business/impl/service/identifiable/entity/parts/WebpageServiceImpl.java

```diff
@@ -111,4 +113,45 @@ public Webpage saveWithParentWebpage(Webpage webpage, UUID parentWebpageUuid)
       throw new IdentifiableServiceException(e.getMessage());
     }
   }
+
+  @Override
+  public BreadcrumbNavigation getBreadcrumbNavigation(UUID uuid) {
+    return ((WebpageRepository) repository).getBreadcrumbNavigation(uuid);
+  }
+
+  @Override
+  public BreadcrumbNavigation getBreadcrumbNavigation(
+      UUID uuid, Locale locale, Locale fallbackLocale) {
+
+    BreadcrumbNavigation localizedBreadcrumbNavigation =
+        ((WebpageRepository) repository).getBreadcrumbNavigation(uuid);
+
+    localizedBreadcrumbNavigation.getNavigationItems().stream()
+        .forEach(
+            n -> {
+              LocalizedText label = n.getLabel();
+
+              // Prepare the fallback solutions, when no label for the desired locale exists.
+              String defaultLabel = label.getText(fallbackLocale);
+              Locale firstLocale =
+                  (label.getLocales() != null && !label.getLocales().isEmpty())
```

Comment:
If there are no locales at all (`label.getLocales() == null`), then the item seems to be skipped. An early return for that case might be easier to understand since there are so many cases to consider.

Reply:
Code has been reworked here and unit tests were added

Reply:
Code has been reworked here and unit tests were added